spec2vec/vector_operations.py

```diff
@@ -60,3 +61,83 @@ def _check_model_coverage():
     weights_raised_tiled = numpy.tile(weights_raised, (1, model.wv.vector_size))
     vector = numpy.sum(word_vectors * weights_raised_tiled, 0)
     return vector
+
+
+@numba.njit
+def cosine_similarity_matrix(vectors_1: numpy.ndarray, vectors_2: numpy.ndarray) -> numpy.ndarray:
+    """Fast implementation of cosine similarity between two arrays of vectors.
+
+    For example:
+
+    .. code-block:: python
+
+        import numpy as np
+        from spec2vec.vector_operations import cosine_similarity_matrix
+
+        vectors_1 = np.array([[1, 1, 0, 0],
+                              [1, 0, 1, 1]])
+        vectors_2 = np.array([[0, 1, 1, 0],
+                              [0, 0, 1, 1]])
+        similarity_matrix = cosine_similarity_matrix(vectors_1, vectors_2)
+
+
+    Parameters
+    ----------
+    vectors_1
+        Numpy array of vectors. vectors_1.shape[0] is number of vectors, vectors_1.shape[1]
+        is vector dimension.
+    vectors_2
+        Numpy array of vectors. vectors_2.shape[0] is number of vectors, vectors_2.shape[1]
+        is vector dimension.
+    """
+    vectors_1 = vectors_1.astype(numpy.float64)  # Numba dot only accepts float or complex arrays
+    vectors_2 = vectors_2.astype(numpy.float64)
+    norm_1 = numpy.sqrt(numpy.sum(vectors_1**2, axis=1))
+    norm_2 = numpy.sqrt(numpy.sum(vectors_2**2, axis=1))
+    for i in range(vectors_1.shape[0]):
+        vectors_1[i] = vectors_1[i] / norm_1[i]
+    for i in range(vectors_2.shape[0]):
+        vectors_2[i] = vectors_2[i] / norm_2[i]
+    return numpy.dot(vectors_1, vectors_2.T)
+
+
+@numba.njit
+def cosine_similarity(vector1: numpy.ndarray, vector2: numpy.ndarray) -> numpy.float64:
+    """Calculate cosine similarity between two input vectors.
+
+    For example:
+
+    .. testcode::
```

Comment:
Good to see a doctest, but doctest is not installed or run on ci.See https://github.com/matchms/matchms/commit/94121c5bbb4926fe958c62dd4b6839a6fc45e148 how to enable. This causes example code not to be rendered in docs.

Reply:
Wow, totally missed that. Thanks.

Reply:
works now